### PR TITLE
PMM-6640 API for HAProxy.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
[PMM-6640](https://jira.percona.com/browse/PMM-6640)

- [x] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1453
- [x] Links to other linked pull requests (optional).
pmm: https://github.com/percona/pmm/pull/667
pmm-api-tests: https://github.com/Percona-Lab/pmm-api-tests/pull/183
---
- [x] Tests passed.
- [ ] Feature build pass.
- [x] (Re)requested review.
- [x] Fix conflicts with target branch.
- [ ] Update API dependency.
- [x] Update jira ticket description if needed.
- [x] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: models/database.go [hunk 1]
```diff
@@ -435,6 +435,20 @@ var databaseSchema = [][]string{
...
 		)`,
 	},
+	28: {
+		`ALTER TABLE services
+			DROP CONSTRAINT services_external_group_check`,
+
+		// Only service with type external or haproxy can have non empty value of group.
+		`ALTER TABLE services
+			ADD CONSTRAINT services_external_group_check_with_haproxy CHECK (
+				((service_type <> '` + string(ExternalServiceType) + `' OR service_type <> '` + string(HAProxyServiceType) + `') AND external_group = '')
```
Comment (askomorokhov) [inline models/database.go:445]:
```suggestion
				((service_type <> '` + string(ExternalServiceType) + `' AND service_type <> '` + string(HAProxyServiceType) + `') AND external_group = '')
```
Shouldn't be here AND?

**Review by askomorokhov — CHANGES_REQUESTED** [review]
Please resolve conflicts.
Do we have an API test for this feature?

File: models/database.go (hunk 1 above)
Comment (JiriCtvrtka) [reply models/database.go:448]: @askomorokhov 
No, it is inside brackets. (external || haproxy) && empty group.
Conflicts are resolved.
Forget about API tests, not meantion in ticket, but I can write them.

File: services/converters.go [hunk 2]
```diff
@@ -166,6 +166,18 @@ func ToAPIService(service *models.Service) (inventorypb.Service, error) {
...
+	case models.HAProxyServiceType:
+		return &inventorypb.HAProxyService{
+			ServiceId:      service.ServiceID,
+			ServiceName:    service.ServiceName,
+			NodeId:         service.NodeID,
+			Environment:    service.Environment,
+			Cluster:        service.Cluster,
+			ReplicationSet: service.ReplicationSet,
+			CustomLabels:   labels,
+			Group:          service.ExternalGroup,
```
Comment (BupycHuk) [inline services/converters.go:178]: let's remove group from everywhere related to HAProxy

File: services/inventory/grpc/services_server.go [hunk 3]
```diff
@@ -217,6 +222,26 @@ func (s *servicesServer) AddExternalService(ctx context.Context, req *inventoryp
...
 
+func (s *servicesServer) AddHAProxyService(ctx context.Context, req *inventorypb.AddHAProxyServiceRequest) (*inventorypb.AddHAProxyServiceResponse, error) {
+	service, err := s.s.AddHAProxyService(ctx, &models.AddDBMSServiceParams{
+		ServiceName:    req.ServiceName,
+		NodeID:         req.NodeId,
+		Environment:    req.Environment,
+		Cluster:        req.Cluster,
+		ReplicationSet: req.ReplicationSet,
+		CustomLabels:   req.CustomLabels,
+		ExternalGroup:  req.Group,
```
Comment (BupycHuk) [inline services/inventory/grpc/services_server.go:233]: ^

File: models/service_model.go [hunk 4]
```diff
@@ -35,6 +35,7 @@ const (
 	PostgreSQLServiceType ServiceType = "postgresql"
 	ProxySQLServiceType   ServiceType = "proxysql"
 	ExternalServiceType   ServiceType = "external"
+	HAProxyServiceType    ServiceType = "haproxy"
```
Comment (BupycHuk) [inline models/service_model.go:38]: let's set haproxy above external here and in other places.

File: services/management/haproxy.go [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+package management
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/percona/pmm-managed/models"
```
Comment (BupycHuk) [inline services/management/haproxy.go:25]: let's fix imports

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: services/converters.go (hunk 2 above)
Comment (JiriCtvrtka) [reply services/converters.go]: Removed.

File: services/inventory/grpc/services_server.go (hunk 3 above)
Comment (JiriCtvrtka) [reply services/inventory/grpc/services_server.go]: Removed.

File: models/service_model.go (hunk 4 above)
Comment (JiriCtvrtka) [reply models/service_model.go:37]: Moved up to external.

File: services/management/haproxy.go (hunk 5 above)
Comment (JiriCtvrtka) [reply services/management/haproxy.go:28]: Ordered.

**Review by BupycHuk — APPROVED** [review]

**Review by askomorokhov — APPROVED** [review]

**Pull request merged**